lib/betagouvbot/notification_schedule.rb

```diff
@@ -18,9 +18,7 @@ def call(members, terms, date)
       private
 
       def date_with_default(date_string)
-        Date.iso8601(date_string)
-      rescue
-        Date.iso8601('3017-01-01')
+        date_string.to_s.empty? || Date.iso8601(date_string)
```

Comment:
Using `rescue` here isn't as problematic in my eye as in the other occasions where I invoked the principle "don't use exceptions for normal control flow" - `Date` does not provide any other mechanism for defaulting to a sensible value.

Here I think the fix introduces two new problems: if date_string is missing, we return `true`, which is not a date; and hence the method name is a lie.